### PR TITLE
[OTLP] Add property-based fuzz testing for ProtobufOtlp serializers

### PR DESCRIPTION
**Pull request by martincostello**
## Changes

Add property-based fuzz testing for the OTLP protobuf serializers for logs, metrics and traces using [FsCheck](https://fscheck.github.io/FsCheck/).

The test code was originally generated by Copilot, but then I refactored it and applied various manual edits, including changes to add coverage for #6771 ~~(as that isn't fixed yet, the fuzz tests should fail)~~.

I've put the fuzzing tests in a new dedicated project, but they can be moved into the same test project as the other tests if desired.

Once there's a new release of OSSF Scorecard containing the changes from https://github.com/ossf/scorecard/pull/4860, this should also improve the score for this repository as it will be detected as fuzzing, which should then address https://github.com/open-telemetry/opentelemetry-dotnet/security/code-scanning/41 which has been open for a while.

~~Depends on #6773.~~

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [ ] ~~Appropriate `CHANGELOG.md` files updated for non-trivial changes~~
* [ ] ~~Changes in public API reviewed (if applicable)~~

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    private static readonly Gen<LogRecordSeverity> LogRecordSeverities = Gen.Elements(
+    [
+        LogRecordSeverity.Debug,
+        LogRecordSeverity.Error,
+        LogRecordSeverity.Fatal,
+        LogRecordSeverity.Info,
+        LogRecordSeverity.Trace,
+        LogRecordSeverity.Unspecified,
+        LogRecordSeverity.Warn,
+    ]);
```
Comment (Kielek) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs:35]: Candidate for separate PR:
```suggestion
    private static readonly Gen<LogRecordSeverity> LogRecordSeverities = Gen.Elements(
    [
        LogRecordSeverity.Debug,
        LogRecordSeverity.Error,
        LogRecordSeverity.Fatal,
        LogRecordSeverity.Info,
        LogRecordSeverity.Trace,
        LogRecordSeverity.Unspecified,
        LogRecordSeverity.Warn,
        (LogRecordSeverity)int.MaxValue,
    ]);
```

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            return Gen.Constant(batch);
+        });
+
+        return gen.ToArbitrary();
+    }
+
+    /// <summary>
+    /// Generates valid LogRecord instances.
+    /// </summary>
+    public static Arbitrary<LogRecord> LogRecordArbitrary()
```
Comment (Kielek) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs]: I think that this method is not used.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+internal static class Generators
+{
+    public static readonly ActivitySource TestActivitySource = new("Fuzz.ActivitySource", "1.0.0");
+
+    private static readonly Gen<ActivityStatusCode> ActivityStatusCodes = Gen.Elements(
+    [
+        ActivityStatusCode.Unset,
+        ActivityStatusCode.Ok,
+        ActivityStatusCode.Error,
+    ]);
```
Comment (Kielek) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs:24]: Candidate for separate PR:
```suggestion
    private static readonly Gen<ActivityStatusCode> ActivityStatusCodes = Gen.Elements(
    [
        ActivityStatusCode.Unset,
        ActivityStatusCode.Ok,
        ActivityStatusCode.Error,
        (ActivityStatusCode)int.MaxValue,
    ]);
```

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs (hunk 1 above)
Comment (martincostello) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs:35]: I'll do those "bad values" as a separate PR, as FsCheck should be able to generate "interesting" values without having to explicitly deal with `MaxValue`, `-1` etc I believe.

Comment (martincostello) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs:35]: Addressed in this PR anyway, as I needed to make some changes locally to use the unused method to generate log records.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs (hunk 2 above)
Comment (martincostello) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs]: Refactored so that it is.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs (hunk 3 above)
Comment (martincostello) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.FuzzTests/Generators.cs:24]: Added a new test for arbitrary statuses of any `int `value.

**Comment by rajkumar-rangaraj** [comment]
Great addition! 🎉 Property-based testing is a good approach for catching edge cases in the serializers.
